doc/source/contributing_code.rst

```diff
@@ -264,16 +285,38 @@ Install the dependencies:
     .. code-block:: bash
 
         python -m pip install \
-          biopython chemfiles codecov cython \
-          griddataformats gsd hypothesis "joblib>=0.12" matplotlib \
-          msmb_theme==1.2.0 netcdf4 networkx "numpy>=1.18.0" \
-          psutil pytest scikit-learn scipy "seaborn>=0.7.0" \
-          sphinx sphinx_rtd_theme "tidynamics>=1.0.0" \
-          "tqdm>=4.43.0"
+	  'Cython>=0.28' \
```

Comment:
It seems a bit superfluous to explicitly pip-install all of this, given that I can just do
```bash
pip install -E package/
```
and then be assured to get the correct versions from pyproject.toml. Should we just recommend users do that? 

This would also reduce the amount of text we have to keep updated.

@RMeli @IAlibay @lilyminium ?

Reply:
Yeah although we probably should do a pip install with all the extras?

Reply:
Yes, just forgot how to do it... was it something like
```
pip install .[extra-formats,analysis,docs]
```

Reply:
`pip install ".[extra-formats,analysis,doc]"` does the trick on my end

Reply:
Pretty sure it's fine to not split but we can deal with it later